src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -59,7 +55,15 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnu
                 console = new AnsiParsingLogConsole();
                 errorConsole = new AnsiParsingLogConsole(stdErr: true);
             }
-            _messageQueue = new ConsoleLoggerProcessor(console, errorConsole);
+            _messageQueue = new ConsoleLoggerProcessor(
+                "LoggingConsoleQueue",
```

Comment:
If there is more than one processor I assume we want to give these unique names. Perhaps add an incrementing number on the end?

Logging, console, and queue are already in the names of the Meter and instrument so we could pick something more compact. How about "provider1", "provider2", etc?

Reply:
There won't be more than one unless it's for testing. Because `ConsoleLoggerProcessor` is an internal class.

Reply:
This is true if there's only one running host in the process which is true for the vast majority of apps using `Microsoft.Extensions.Logging.Console`. It is possible to run multiple hosts in a process though even in non-testing scenarios. I'm not sure how important we consider that to be.

Reply:
By more than one, I mean more than one in the entire process. Its true there will only be one Processor per ConsoleLoggerProvider, but ConsoleLoggerProvider is a public type and users can (and do) create multiple of these in the same process. As a test scenario create two LoggerFactories, add a new ConsoleLoggerProvider() to each one, and then log to both of them.

Reply:
Fixed the code to properly support metrics when multiple console logger providers are present:

![image](https://user-images.githubusercontent.com/5897654/172692969-9cdb75b9-9691-4adf-b5c9-4bfdedae44af.png)

To get this I instantiated 3 different `LoggerFactory` instances each with their own console provider and processor as you described.